crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -117,40 +107,60 @@ impl TextureAtlas {
 
         let grid_size = Vec2::new(columns as f32, rows as f32);
 
-        TextureAtlas {
+        Self {
             size: ((tile_size + current_padding) * grid_size) - current_padding,
             textures: sprites,
-            texture,
             texture_handles: None,
         }
     }
 
-    /// Add a sprite to the list of textures in the [`TextureAtlas`]
-    /// returns an index to the texture which can be used with [`TextureAtlasSprite`]
+    /// Add a *section* to the list in the layout and returns its index
+    /// which can be used with [`TextureAtlas`]
     ///
     /// # Arguments
     ///
-    /// * `rect` - The section of the atlas that contains the texture to be added,
-    /// from the top-left corner of the texture to the bottom-right corner
+    /// * `rect` - The section of the texture to be added
+    ///
+    /// [`TextureAtlas`]: crate::TextureAtlas
     pub fn add_texture(&mut self, rect: Rect) -> usize {
         self.textures.push(rect);
         self.textures.len() - 1
     }
 
-    /// The number of textures in the [`TextureAtlas`]
+    /// How many textures are in the `TextureAtlas`
     pub fn len(&self) -> usize {
         self.textures.len()
     }
 
-    /// Returns `true` if there are no textures in the [`TextureAtlas`]
     pub fn is_empty(&self) -> bool {
         self.textures.is_empty()
     }
 
-    /// Returns the index of the texture corresponding to the given image handle in the [`TextureAtlas`]
+    /// Retrieves the texture *section* index of the given `texture` handle.
+    ///
+    /// This requires the layout to have been built using a [`TextureAtlasBuilder`]
+    ///
+    /// [`TextureAtlasBuilder`]: crate::TextureAtlasBuilder
     pub fn get_texture_index(&self, texture: &Handle<Image>) -> Option<usize> {
         self.texture_handles
             .as_ref()
             .and_then(|texture_handles| texture_handles.get(texture).cloned())
     }
 }
+
+impl TextureAtlas {
+    /// Retrieves the current texture [`Rect`] of the sprite sheet according to the section `index`
+    pub fn texture_rect(&self, texture_atlases: &Assets<TextureAtlasLayout>) -> Option<Rect> {
```

Comment:
This function returns None in both cases where there's no found atlas layout or no found rect within the layout, right? Could this be problematic when debugging code?

Reply:
I doubt it may be an issue but we could reserve it to inner use with `pub(crate)` what do you think @alice-i-cecile ?

Reply:
I think keep it public and expand the docs for now.